DashKitUI/ViewControllers/PodSettingsSetupViewControllers.swift

```diff
@@ -0,0 +1,204 @@
+//  DashKitUI
+//
+//  Created by Pete Schwamb on 9/25/18.
+//  Copyright © 2018 Pete Schwamb. All rights reserved.
+//
+
+import Foundation
+
+import UIKit
+import HealthKit
+import LoopKit
+import LoopKitUI
+import DashKit
+
+class PodSettingsSetupViewController: SetupTableViewController {
+
+    private var pumpManagerSetupViewController: DashPumpManagerSetupViewController? {
+        return navigationController as? DashPumpManagerSetupViewController
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        updateContinueButton()
+
+        tableView.register(SettingsTableViewCell.self, forCellReuseIdentifier: SettingsTableViewCell.className)
+    }
+
+    fileprivate lazy var quantityFormatter: QuantityFormatter = {
+        let quantityFormatter = QuantityFormatter()
+        quantityFormatter.numberFormatter.minimumFractionDigits = 0
+        quantityFormatter.numberFormatter.maximumFractionDigits = 3
+
+        return quantityFormatter
+    }()
+
+    func updateContinueButton() {
+        let enabled: Bool
+        if pumpManagerSetupViewController?.maxBolusUnits == nil || pumpManagerSetupViewController?.maxBasalRateUnitsPerHour == nil {
+            enabled = false
+        }
+        else if let basalSchedule = pumpManagerSetupViewController?.basalSchedule {
+            enabled = !basalSchedule.items.isEmpty && !scheduleHasError
+        } else {
+            enabled = false
+        }
+        footerView.primaryButton.isEnabled = enabled
+    }
+
+    var scheduleHasError: Bool {
+        return scheduleErrorMessage != nil
+    }
+
+    var scheduleErrorMessage: String? {
+        if let basalRateSchedule = pumpManagerSetupViewController?.basalSchedule {
+            if basalRateSchedule.items.count > Pod.maximumBasalScheduleEntryCount {
+                return LocalizedString("Too many entries", comment: "The error message shown when Loop's basal schedule has more entries than the pod can support")
+            }
+            let allowedRates = Pod.supportedBasalRates
+            if basalRateSchedule.items.contains(where: {!allowedRates.contains($0.value)}) {
+                return LocalizedString("Invalid entry", comment: "The error message shown when Loop's basal schedule has an unsupported rate")
+            }
+        }
+        return nil
+    }
+
+    // MARK: - Table view data source
+
+    private enum Section: Int, CaseIterable {
+        case description
+        case configuration
+    }
+
+    private enum ConfigurationRow: Int, CaseIterable {
+        case deliveryLimits
+        case basalRates
+    }
+
+    override func numberOfSections(in tableView: UITableView) -> Int {
+        return Section.allCases.count
+    }
+
+    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        switch Section(rawValue: section)! {
+        case .description:
+            return 1
+        case .configuration:
+            return 2
```

Comment:
Should this be `allCases` of the `ConfigurationRow` enum?

Reply:
Yep